src/layout/IHyprLayout.cpp

```diff
@@ -392,141 +392,141 @@ void IHyprLayout::onEndDragWindow() {
     g_pInputManager->m_bWasDraggingWindow = false;
 }
 
-static inline bool canSnap(const double sideA, const double sideB, const double gap) {
-    return std::abs(sideA - sideB) < gap;
+static inline bool canSnap(const double SIDEA, const double SIDEB, const double GAP) {
+    return std::abs(SIDEA - SIDEB) < GAP;
 }
 
-static void snapMoveLeft(double& pos, double& len, const double p) {
-    pos = p;
+static void snapMove(double& start, double& end, const double P) {
+    end   = P + (end - start);
+    start = P;
 }
 
-static void snapMoveRight(double& pos, double& len, const double p) {
-    pos = p - len;
-}
-
-static void snapResizeLeft(double& pos, double& len, const double p) {
-    len += pos - p;
-    pos = p;
-}
-
-static void snapResizeRight(double& pos, double& len, const double p) {
-    len = p - pos;
+static void snapResize(double& start, double& end, const double P) {
+    start = P;
 }
 
 typedef std::function<void(double&, double&, const double)> SnapFn;
 
-static void performSnap(Vector2D& sourcePos, Vector2D& sourceSize, PHLWINDOW DRAGGINGWINDOW, const eMouseBindMode mode, const int corner, const Vector2D& beginSize) {
-    static auto  SNAPWINDOWGAP  = CConfigValue<Hyprlang::INT>("general:snap:window_gap");
-    static auto  SNAPMONITORGAP = CConfigValue<Hyprlang::INT>("general:snap:monitor_gap");
+static void performSnap(Vector2D& sourcePos, Vector2D& sourceSize, PHLWINDOW DRAGGINGWINDOW, const eMouseBindMode MODE, const int CORNER, const Vector2D& BEGINSIZE) {
+    static auto  SNAPWINDOWGAP     = CConfigValue<Hyprlang::INT>("general:snap:window_gap");
+    static auto  SNAPMONITORGAP    = CConfigValue<Hyprlang::INT>("general:snap:monitor_gap");
+    static auto  SNAPBORDEROVERLAP = CConfigValue<Hyprlang::INT>("general:snap:border_overlap");
 
-    const SnapFn snapRight = (mode == MBIND_MOVE) ? snapMoveRight : snapResizeRight;
-    const SnapFn snapLeft  = (mode == MBIND_MOVE) ? snapMoveLeft : snapResizeLeft;
-    const SnapFn snapDown  = snapRight;
-    const SnapFn snapUp    = snapLeft;
-    int          snaps     = 0;
+    const SnapFn SNAP  = (MODE == MBIND_MOVE) ? snapMove : snapResize;
+    int          snaps = 0;
 
+    const bool   OVERLAP            = *SNAPBORDEROVERLAP;
     const int    DRAGGINGBORDERSIZE = DRAGGINGWINDOW->getRealBorderSize();
-    CBox         sourceBox          = CBox{sourcePos, sourceSize}.expand(DRAGGINGBORDERSIZE);
+
+    struct Range {
```

Comment:
SRange

Reply:
I can change the name, but what does the 'S' stand for in this context?

Reply:
STRUCT